diff -Nur vte-0.11.11.orig/configure.in vte-0.11.11/configure.in
--- vte-0.11.11.orig/configure.in	2004-05-12 01:05:48.000000000 +0200
+++ vte-0.11.11/configure.in	2004-05-12 02:15:16.000000000 +0200
@@ -17,7 +17,7 @@
 # On some OSs, AC_PATH_XTRA doesn't work right(?), so let the user specify
 # X_PRE_LIBS and X_EXTRA_LIBS to add even more libraries, and add -lX11 to
 # the list of libraries for grins.
-AC_PATH_XTRA
+AC_PATH_XTRA( [X11], [X11/Xlib.h], [XSync ()])
 X_CFLAGS=
 X_LIBS=
 if test "$have_x" = yes ; then
