GNOME #112172; move VteReaper down the library stack

diff -Nur -x '*.orig' -x '*~' vte-0.19.4/python/vte.defs vte-0.19.4.new/python/vte.defs
--- vte-0.19.4/python/vte.defs	2008-10-23 07:11:06.000000000 +0100
+++ vte-0.19.4.new/python/vte.defs	2009-01-11 23:03:08.000000000 +0000
@@ -7,6 +7,14 @@
   (gtype-id "VTE_TYPE_TERMINAL")
 )
 
+(define-object Reaper
+  (in-module "Vte")
+  (parent "GObject")
+  (c-name "VteReaper")
+  (gtype-id "VTE_TYPE_REAPER")
+)
+
+
 ;; Enumerations and flags ...
 
 (define-enum TerminalEraseBinding
@@ -690,6 +698,17 @@
   )
 )
 
+;; From ../src/reaper.h
+(define-function vte_reaper_get_type
+  (c-name "vte_reaper_get_type")
+  (return-type "GtkType")
+)
+
+(define-function reaper_get
+  (c-name "vte_reaper_get")
+  (return-type "VteReaper*")
+)
+
 (define-method get_encoding
   (of-object "VteTerminal")
   (c-name "vte_terminal_get_encoding")
diff -Nur -x '*.orig' -x '*~' vte-0.19.4/python/vtemodule.c vte-0.19.4.new/python/vtemodule.c
--- vte-0.19.4/python/vtemodule.c	2008-11-29 21:57:09.000000000 +0000
+++ vte-0.19.4.new/python/vtemodule.c	2009-01-11 23:03:58.000000000 +0000
@@ -25,6 +25,7 @@
 #include <pygobject.h>
 #include <pygtk/pygtk.h>
 #include "../src/vte.h"
+#include "../src/reaper.h"
 
 extern void pyvte_register_classes(PyObject * d);
 extern void pyvte_add_constants(PyObject *module, const gchar *strip_prefix);
diff -Nur -x '*.orig' -x '*~' vte-0.19.4/python/vte.override vte-0.19.4.new/python/vte.override
--- vte-0.19.4/python/vte.override	2008-11-22 14:51:06.000000000 +0000
+++ vte-0.19.4.new/python/vte.override	2009-01-11 23:03:37.000000000 +0000
@@ -7,6 +7,7 @@
 #include <pygobject.h>
 #include <gtk/gtk.h>
 #include "../src/vte.h"
+#include "../src/reaper.h"
 
 void pyvte_add_constants(PyObject *module, const gchar *strip_prefix);
 void pyvte_register_classes(PyObject *d);
@@ -16,6 +17,7 @@
 import gtk.gdk.Pixbuf as PyGdkPixbuf_Type
 import gtk.MenuShell as PyGtkMenuShell_Type
 import gtk.Widget as PyGtkWidget_Type
+import gobject.GObject as PyGObject_Type
 %%
 override vte_terminal_feed kwargs
 static PyObject *
