diff -Nur -x '*.orig' -x '*~' vte-0.19.4/configure.in vte-0.19.4.new/configure.in
--- vte-0.19.4/configure.in	2008-12-15 19:55:59.000000000 +0000
+++ vte-0.19.4.new/configure.in	2009-01-12 22:05:19.000000000 +0000
@@ -289,14 +289,25 @@
 AC_CHECK_FUNC(floor,,AC_CHECK_LIB(m,floor,LIBS=["$LIBS -lm"]))
 AC_CHECK_FUNCS([ceil floor])
 
+AC_ARG_WITH(ncurses,
+	    [AS_HELP_STRING([--without-ncurses],
+	 	            [don't use ncurses library])
+	    ], with_ncurses=$withval, with_ncurses=yes)
+
 # Look for tgetent
 
-AC_CHECK_FUNC([tgetent],[vte_cv_termcap_lib=libc],
-  [AC_CHECK_LIB([ncurses],[tgetent],[vte_cv_termcap_lib=libncurses],
-    [AC_CHECK_LIB([tinfo],[tgetent],[vte_cv_termcap_lib=libtinfo],
-      [AC_CHECK_LIB([curses],[tgetent],[vte_cv_termcap_lib=libcurses],
-        [AC_CHECK_LIB([termcap],[tgetent],[vte_cv_termcap_lib=libtermcap],
-          [vte_cv_termcap_lib=])])])])])
+if test x$with_ncurses = xyes ; then
+  AC_CHECK_FUNC([tgetent],[vte_cv_termcap_lib=libc],
+    [AC_CHECK_LIB([ncurses],[tgetent],[vte_cv_termcap_lib=libncurses],
+      [AC_CHECK_LIB([tinfo],[tgetent],[vte_cv_termcap_lib=libtinfo],
+        [AC_CHECK_LIB([curses],[tgetent],[vte_cv_termcap_lib=libcurses],
+          [AC_CHECK_LIB([termcap],[tgetent],[vte_cv_termcap_lib=libtermcap],
+            [vte_cv_termcap_lib=])])])])])
+else
+  AC_CHECK_FUNC([tgetent],[vte_cv_termcap_lib=libc],
+    [AC_CHECK_LIB([termcap],[tgetent],[vte_cv_termcap_lib=libtermcap],
+      [vte_cv_termcap_lib=])])
+fi
 
 case "$vte_cv_termcap_lib" in
   libc)       # FIXME: which headers to include here?
