--- vte-0.16.14/configure.in	2008-08-03 16:38:00.000000000 +0200
+++ vte-0.16.14.new/configure.in	2008-08-03 16:38:45.000000000 +0200
@@ -296,16 +296,23 @@
 AC_CHECK_FUNC(floor,,AC_CHECK_LIB(m,floor,LIBS=["$LIBS -lm"]))
 AC_CHECK_FUNCS([ceil floor])
 
-# Look for ncurses or curses or termcap.
-AC_CHECK_HEADER(ncurses.h,[AC_CHECK_HEADER(term.h,[AC_CHECK_LIB(ncurses,tgetent,[LIBS="-lncurses $LIBS";AC_DEFINE(HAVE_NCURSES,1,[Define if you have ncurses.h and libncurses])])])])
-if test x$ac_cv_lib_ncurses_tgetent != xyes ; then
-	AC_CHECK_HEADER(ncurses/curses.h,[AC_CHECK_HEADER(ncurses/term.h,[AC_CHECK_LIB(ncurses,tgetent,[LIBS="-lncurses $LIBS";AC_DEFINE(HAVE_NCURSES_CURSES,1,[Define if you have ncurses/curses.h and libncurses])])])])
+AC_ARG_WITH(ncurses,
+	    [AS_HELP_STRING([--without-ncurses],
+	 	            [don't use ncurses library])
+	    ], with_ncurses=$withval, with_ncurses=yes)
+
+if test x$with_ncurses = xyes ; then
+	# Look for ncurses or curses or termcap.
+	AC_CHECK_HEADER(ncurses.h,[AC_CHECK_HEADER(term.h,[AC_CHECK_LIB(ncurses,tgetent,[LIBS="-lncurses $LIBS";AC_DEFINE(HAVE_NCURSES,1,[Define if you have ncurses.h and libncurses])])])])
 	if test x$ac_cv_lib_ncurses_tgetent != xyes ; then
-		AC_CHECK_HEADER(curses.h,[AC_CHECK_HEADER(term.h,[AC_CHECK_LIB(curses,tgetent,[LIBS="-lcurses $LIBS";AC_DEFINE(HAVE_CURSES,1,[Define if you have curses.h and libcurses])])])])
-		if test x$ac_cv_lib_curses_tgetent != xyes ; then
-			AC_CHECK_HEADER(termcap.h,[AC_CHECK_LIB(termcap,tgetent,[LIBS="-ltermcap $LIBS";AC_DEFINE(HAVE_TERMCAP,1,[Define if you have termcap.h and libtermcap])])])
-			if test x$ac_cv_header_termcap_h != xyes ; then
-				AC_MSG_ERROR([You must have at least one of ncurses, curses, or libtermcap installed to build vte.])
+		AC_CHECK_HEADER(ncurses/curses.h,[AC_CHECK_HEADER(ncurses/term.h,[AC_CHECK_LIB(ncurses,tgetent,[LIBS="-lncurses $LIBS";AC_DEFINE(HAVE_NCURSES_CURSES,1,[Define if you have ncurses/curses.h and libncurses])])])])
+		if test x$ac_cv_lib_ncurses_tgetent != xyes ; then
+			AC_CHECK_HEADER(curses.h,[AC_CHECK_HEADER(term.h,[AC_CHECK_LIB(curses,tgetent,[LIBS="-lcurses $LIBS";AC_DEFINE(HAVE_CURSES,1,[Define if you have curses.h and libcurses])])])])
+			if test x$ac_cv_lib_curses_tgetent != xyes ; then
+				AC_CHECK_HEADER(termcap.h,[AC_CHECK_LIB(termcap,tgetent,[LIBS="-ltermcap $LIBS";AC_DEFINE(HAVE_TERMCAP,1,[Define if you have termcap.h and libtermcap])])])
+				if test x$ac_cv_header_termcap_h != xyes ; then
+					AC_MSG_ERROR([You must have at least one of ncurses, curses, or libtermcap installed to build vte.])
+				fi
 			fi
 		fi
 	fi
