diff -Nur vte-0.16.0/src/vte.c vte-0.16.0.new/src/vte.c
--- vte-0.16.0/src/vte.c	2007-03-08 17:47:15.000000000 +0100
+++ vte-0.16.0.new/src/vte.c	2007-03-13 23:04:34.000000000 +0100
@@ -9773,12 +9773,10 @@
 			row * height + VTE_PAD_WIDTH,
 			(col_stop - col) * width,
 			(row_stop - row) * height);
-	if (!GTK_WIDGET_DOUBLE_BUFFERED (terminal) ||
-			_vte_draw_has_background_image (terminal->pvt->draw)) {
-		_vte_draw_clear (terminal->pvt->draw,
-				area->x, area->y,
-				area->width, area->height);
-	}
+
+	_vte_draw_clear (terminal->pvt->draw,
+			 area->x, area->y,
+			 area->width, area->height);
 
 	/* Now we're ready to draw the text.  Iterate over the rows we
 	 * need to draw. */
