diff -Nur vte-0.16.3/acinclude.m4 vte-0.16.3.new/acinclude.m4
--- vte-0.16.3/acinclude.m4	2007-02-26 22:38:00.000000000 +0100
+++ vte-0.16.3.new/acinclude.m4	2007-05-04 11:55:19.000000000 +0200
@@ -9,13 +9,15 @@
 dnl deduce PYTHON_INCLUDES
 py_prefix=`$PYTHON -c "import sys; print sys.prefix"`
 py_exec_prefix=`$PYTHON -c "import sys; print sys.exec_prefix"`
+if $PYTHON-config --help 2>/dev/null; then
+PYTHON_INCLUDES=`$PYTHON-config --includes 2>/dev/null`
+else
 PYTHON_INCLUDES="-I${py_prefix}/include/python${PYTHON_VERSION}"
 if test "$py_prefix" != "$py_exec_prefix"; then
   PYTHON_INCLUDES="$PYTHON_INCLUDES -I${py_exec_prefix}/include/python${PYTHON_VERSION}"
 fi
-PYTHON_LIBS="-L${py_prefix}/libs -lpython${PYTHON_VERSION}"
+fi
 AC_SUBST(PYTHON_INCLUDES)
-AC_SUBST(PYTHON_LIBS)
 dnl check if the headers exist:
 save_CPPFLAGS="$CPPFLAGS"
 CPPFLAGS="$CPPFLAGS $PYTHON_INCLUDES"
@@ -43,7 +45,7 @@
 dnl specific counterparts like ac_cxx_rtti.m4 that will test for
 dnl -fno-rtti -fno-exceptions
 dnl 
-dnl @version $Id: acinclude.m4 1763 2007-02-26 21:34:01Z cpwilson $
+dnl @version $Id: acinclude.m4 1077 2004-04-20 05:18:45Z nalin $
 dml @author  Guido Draheim <guidod@gmx.de>
 
 AC_DEFUN([AC_CHECK_CC_OPT],
