Index: src/vte.c
===================================================================
--- src/vte.c	(revision 1834)
+++ src/vte.c	(working copy)
@@ -1967,6 +1967,7 @@
 
 	screen = terminal->pvt->screen;
 
+	_vte_terminal_ensure_row (terminal);
 	/* The total number of lines.  Add one to the cursor offset
 	 * because it's zero-based. */
 	rows = MAX(_vte_ring_next(terminal->pvt->screen->row_data),
