--- trunk/src/vte.c	2007/02/27 12:16:37	1770
+++ trunk/src/vte.c	2007/02/28 10:18:32	1776
@@ -7381,7 +7381,8 @@
 		/* Ensure the cursor is valid */
 		screen->cursor_current.row = CLAMP (screen->cursor_current.row,
 				_vte_ring_delta (screen->row_data),
-				_vte_ring_next (screen->row_data) - 1);
+				MAX (_vte_ring_delta (screen->row_data),
+					_vte_ring_next (screen->row_data) - 1));
 		/* Notify viewers that the contents have changed. */
 		_vte_terminal_queue_contents_changed(terminal);
 	}
