diff -Nur vte-0.15.2/src/pty.c vte-0.15.2.new/src/pty.c
--- vte-0.15.2/src/pty.c	2007-02-12 16:31:46.000000000 +0100
+++ vte-0.15.2.new/src/pty.c	2007-02-12 16:32:11.000000000 +0100
@@ -276,7 +276,8 @@
 
 	if (command != NULL) {
 		gchar **arg2;
-		gint i, argc;
+		gchar **envp2 = NULL;
+		gint i, k, envc, argc;
 
 		/* push the command into argv[0] */
 		argc = argv ? g_strv_length (argv) : 0;
@@ -297,9 +298,19 @@
 #endif
 		}
 		arg2[i+1] = NULL;
-
+		// envp is environemnt *added* according to the vte
+		// manual, so deal with that here and now
+		if(envp != NULL) {
+		   envc = g_strv_length(environ)+g_strv_length(envp);
+		   envp2 = g_new(char *, envc+2);
+		   for(i=0;environ[i]!=NULL;i++) 
+		      envp2[i] = g_strdup(environ[i]);
+		   for(k=0;envp[k]!=NULL;k++,i++)
+		      envp2[i] = g_strdup(envp[k]);
+		   envp2[i] = NULL;
+		}
 		ret = g_spawn_async_with_pipes (directory,
-				arg2, envp,
+				arg2, envp2, 
 				G_SPAWN_CHILD_INHERITS_STDIN |
 				G_SPAWN_FILE_AND_ARGV_ZERO |
 				G_SPAWN_SEARCH_PATH |
@@ -309,6 +320,7 @@
 				NULL, NULL, NULL,
 				error);
 		g_strfreev (arg2);
+		g_strfreev (envp2);
 	}
 #ifdef HAVE_FORK
 	else {
