diff -Nur vte-0.16.0/src/vte.c vte-0.16.0.new/src/vte.c
--- vte-0.16.0/src/vte.c	2007-03-15 19:00:44.000000000 +0200
+++ vte-0.16.0.new/src/vte.c	2007-03-15 19:00:51.000000000 +0200
@@ -10103,18 +10103,6 @@
 							FALSE,
 							TRUE,
 							width, height);
-			} else
-			if (preedit_cursor == len) {
-				/* Empty cursor at the end. */
-				vte_terminal_draw_cells(terminal,
-							&items[len], 1,
-							back, fore, TRUE, TRUE,
-							FALSE,
-							FALSE,
-							FALSE,
-							FALSE,
-							FALSE,
-							width, height);
 			}
 			g_free(items);
 		}
