Index: vte-0.27.2/src/vtebg.c
===================================================================
--- vte-0.27.2.orig/src/vtebg.c	2010-12-13 16:14:54.627609272 -0500
+++ vte-0.27.2/src/vtebg.c	2010-12-13 16:15:16.467609272 -0500
@@ -240,7 +240,11 @@
 
 		window = gdk_screen_get_root_window(screen);
                 pvt->native.window = window;
+#if GTK_CHECK_VERSION (2, 91, 6)
+                pvt->native.native_window = gdk_x11_window_get_xid(window);
+#else
                 pvt->native.native_window = gdk_x11_drawable_get_xid(window);
+#endif
 #if GTK_CHECK_VERSION (2, 90, 8)
                 pvt->native.display = gdk_window_get_display(window);
 #else
